climada_petals/engine/supplychain.py

```diff
@@ -604,8 +711,15 @@ def calc_impacts(self,
 
         self.calc_matrices(io_approach=io_approach)
 
+        # This block is a temporary `fix`, before a deeper rework of the module.
+        # The underlying problem is that if secs_shock was set outside this method,
+        # impact and exposure are ignored as ``calc_shock_to_sectors`` is not run in the next block.
+        # The objective here is to warn user.
+        if (exposure is not None) or (impact is not None) and self.secs_shock is not None:
```

Comment:
don't you need a bracket here? ( ... or ...) and ...
"and" has higher precedence than "or"

Reply:
Indeed, nice catch!